lua/kubectl/utils/tables.lua

```diff
@@ -386,12 +386,12 @@ end
 function M.find_resource(data, name, namespace)
   if data.items then
     return vim.iter(data.items):find(function(row)
-      return row.metadata.name == name and row.metadata.namespace == namespace
+      return row.metadata.name == name and (namespace and row.metadata.namespace == namespace or true)
     end)
   end
   if data.rows then
     return vim.iter(data.rows):find(function(row)
-      return row.object.metadata.name == name and row.object.metadata.namespace == namespace
+      return row.object.metadata.name == name and (namespace and row.object.metadata.namespace == namespace or true)
```

Comment:
Am I thinking wrong but won't this always return true:
`(namespace and row.object.metadata.namespace == namespace or true)`

Reply:
I tested with this file:

```lua
---@diagnostic disable: undefined-global
local M = {
	completions = {},
}

-- Reload the file when it changes on disk
local group = vim.api.nvim_create_augroup("ReloadModule", { clear = true })
vim.api.nvim_create_autocmd("BufWritePost", {
	buffer = 0,
	group = group,
	callback = function()
		vim.cmd("luafile %")
		vim.notify("Module reloaded")
	end,
})

vim.keymap.set("n", "bla", function()
	-- Define the user command

	local data = {
		rows = {
			{
				object = {
					metadata = {
						name = "test",
						namespace = "default",
					},
				},
			},
			{
				object = {
					metadata = {
						name = "no-ns",
					},
				},
			},
		},
	}
	local name = "test"
	local namespace = "default"
	-- local name = "no-ns"
	local hiri = vim.iter(data.rows):find(function(row)
		return row.object.metadata.name == name and (namespace and row.object.metadata.namespace == namespace or true)
	end)
	vim.notify(vim.inspect(hiri))
end)
```

- save the file
- open it
- run `:so%` to source the file
- click `bla` to trigger the function - now you see the object with name and namespace
- comment out name and namespace
- comment in no-ns
- save (it will reload the module)
- click `bla` - now you see the object without the namespace

Maybe it's not the smartest(?) but it's working